lib/equalizer.rb

```diff
@@ -9,84 +9,34 @@ class Equalizer < Module
   #
   # @param [Array<Symbol>] keys
   #
-  # @return [undefined]
+  # @return [Module]
   #
-  # @api private
-  def initialize(*keys)
-    @keys = keys
-    define_methods
-    freeze
+  # @api public
+  def self.new(*keys)
+    Module.new do
+      define_method(:keys) { keys }
```

Comment:
This infects the includer of the module with a method `keys`. The whole idea is that Equalizer only adds `#==` and friends but no other side effects.